### ISSUE TITLE
Problems with new `buildkite-agent` module

### ISSUE THREAD

**Issue by dhess** [LnL7/nix-darwin#657]
The new `buildkite-agent` module doesn't activate properly, at least on our `nix-darwin`-managed Macs. In our configs, at least, the `nix-darwin` activation script won't create new users & groups unless they're in the `users.knownUsers` and `users.knownGroups` lists; see, for example, the `ofborg` module in this project. The new module doesn't do that. Additionally, when users/groups are added to those lists, they also require a fixed `uid` and `gid`, respectively, which the new module also doesn't do.

I've never understood precisely what & why the `users.{knownUsers,knownGroups}` lists are for, so it's possible that something in our `nix-darwin` configs is non-standard, but presumably at least some other users will also need these values to be set if they want to use the `buildkite-agent` module.

**Comment by dhess** [comment]
The `buildkite-agent` `launchd` config should also probably set the `NIX_SSL_CERT_FILE` environment variable to `"${pkgs.cacert}/etc/ssl/certs/ca-bundle.crt"`. Otherwise, `git clone` on a public `git` repo will die with a cryptic error message:

```sh
$ git clone -v -- https://github.com/hackworthltd/hacknix.git .
Cloning into '.'...
fatal: unable to access 'https://github.com/hackworthltd/hacknix.git/': OpenSSL/3.0.8: error:16000069:STORE routines::unregistered scheme
```

This one took awhile to figure out.

**Comment by rvem** [comment]
> The buildkite-agent launchd config should also probably set the NIX_SSL_CERT_FILE environment variable to "${pkgs.cacert}/etc/ssl/certs/ca-bundle.crt"

Indeed, looks like I accidentally dropped `NIX_SSL_CERT_FILE` from the environment, while migrating to the new module structure.
Should be fixed by #675

**Comment by rvem** [comment]
As for groups/users setup, the old version didn't add them to `known{Groups, Users}` as well
